src/frontegg-mock/src/lib.rs

```diff
@@ -637,6 +642,103 @@ async fn handle_delete_user(
     }
 }
 
+// https://docs.frontegg.com/reference/userscontrollerv3_getusers
+async fn handle_get_users_v3(
+    State(context): State<Arc<Context>>,
+    Query(query): Query<UsersV3Query>,
+) -> Result<Json<UsersV3Response>, StatusCode> {
+    let users = context.users.lock().unwrap();
+    let role_mapping: BTreeMap<String, UserRole> = context
+        .roles
+        .iter()
+        .map(|role| (role.id.clone(), role.clone()))
+        .collect();
+
+    let mut filtered_users: Vec<UserResponse> = users
+        .iter()
+        .filter(|(email, user)| {
+            query
+                .email
+                .as_ref()
+                .map_or(true, |q_email| *email == q_email)
+                && query.ids.as_ref().map_or(true, |ids| {
+                    ids.split(',').any(|id| id == user.id.to_string())
+                })
+        })
+        .map(|(_, user)| UserResponse {
+            id: user.id,
+            email: user.email.clone(),
+            verified: user.verified.unwrap_or(true),
+            metadata: user.metadata.clone().unwrap_or_default(),
+            provider: user.auth_provider.clone().unwrap_or_default(),
+            roles: get_user_roles(&user.roles, &role_mapping),
+        })
+        .collect();
+
+    // Sort users if sort_by is provided
+    if let Some(sort_by) = &query.sort_by {
+        filtered_users.sort_by(|a, b| {
+            let cmp = match sort_by.as_str() {
+                "email" => a.email.cmp(&b.email),
+                "id" => a.id.cmp(&b.id),
+                _ => std::cmp::Ordering::Equal,
+            };
+            if query.order.as_deref() == Some("desc") {
```

Comment:
In their docs, this is all caps. Perhaps this order should be an enum, rather than doing string matching?